mlflow/recipes/steps/split.py

```diff
@@ -38,6 +42,47 @@ def _make_elem_hashable(elem):
         return elem
 
 
+def _run_split(task, input_df, split_ratios, target_col):
+    if task == "classification":
+        return _perform_split_per_class(input_df, split_ratios, target_col)
```

Comment:
Using the current split implementation for multi-class problems we might run into the situation that the test or validation dataset will not get any examples of some heavily under-represented classes. This primarily affects multi-class classification but can also affect binary if we consider anomaly detection or fraud use cases where the positive (fraud) class is usually heavily under-represented. Our rebalancing feature cannot help here since we apply it to the train dataset only. As a result, users can get errors during the evaluation step because mlflow.evaluate() checks if all classes are represented in the test/validation dataset.

Reply:
I have changed the function name accordingly.

Reply:
>Stratification is a nice feature to have but is it always desired?
I do not see any drawbacks to classification use cases..